featuretools/primitives/utils.py

```diff
@@ -134,6 +204,94 @@ def _get_descriptions(primitives):
     return descriptions
 
 
+def _get_summary_primitives(primitives: List) -> Dict[str, int]:
+    """Provides metrics for a list of primatives."""
+    unique_input_types = set()
+    unique_output_types = set()
+    ct_multi_in = 0
+    ct_multi_out = 0
+    ct_extra_data = 0
+    ct_controllable = 0
+    # ct_trainable = 0  # get clarification
+    prim_input_ww_checks = {  # semantic tag for time index and logical types in inputs
+        "semantic_time_idx": 0,
+        "Datetime": 0,
+        "Categorical": 0,
+        "Address": 0,
+        "Age": 0,
+        "AgeFractional": 0,
+        "AgeNullable": 0,
+        "Boolean": 0,
+        "BooleanNullable": 0,
+        "CountryCode": 0,
+        "CurrencyCode": 0,
+        "Double": 0,
+        "Integer": 0,
+        "IntegerNullable": 0,
+        "EmailAddress": 0,
+        "Filepath": 0,
+        "PersonFullName": 0,
+        "IPAddress": 0,
+        "LatLong": 0,
+        "NaturalLanguage": 0,
+        "Unknown": 0,
+        "Ordinal": 0,
+        "PhoneNumber": 0,
+        "SubRegionCode": 0,
+        "Timedelta": 0,
+        "URL": 0,
+        "PostalCode": 0,
+    }
+    for prim in primitives:
+        input_checks = set()
+        _check_input_types(prim.input_types, input_checks, unique_input_types)
+        for check in list(input_checks):
+            prim_input_ww_checks[check] += 1
+
+        if len(prim.input_types) > 1:
+            ct_multi_in += 1
+
+        if isinstance(prim.return_type, list):
+            for out_type in prim.return_type:
+                unique_output_types.add(str(out_type))
+                ct_multi_out += 1
+        else:
+            unique_output_types.add(str(prim.return_type))
+
+        if hasattr(prim, "filename"):
+            ct_extra_data += 1
+
+        if len(getfullargspec(prim.__init__).args) > 1:
+            ct_controllable += 1
+
+    return {
+        "unique_input_types": len(unique_input_types),
+        "unique_output_types": len(unique_output_types),
+        "ct_multi_in": ct_multi_in,
+        "ct_multi_out": ct_multi_out,
+        "ct_extra_data": ct_extra_data,
+        "ct_controllable": ct_controllable,
+        **prim_input_ww_checks,
+    }
+
+
+def _check_input_types(
```

Comment:
I wonder if we should create a generic utility function that pulls out and flattens input types?

@tamargrey what do you think?

Reply:
Yeah, I was wondering about this too since it's similar to the `_get_unique_input_types` function. But this function changes everything to string. I guess one way we could do it is to have these functions call a flatten method that just outputs a list of all input types. The dev can then choose what they want to do with it (throw it into a set(), change it to string etc...)

Reply:
@ozzieD i like that idea!

Reply:
done